iree/modules/strings/BUILD

```diff
@@ -7,6 +7,15 @@ package(
     licenses = ["notice"],  # Apache 2.0
 )
 
+iree_cmake_extra_content(
+    content = """
+if (NOT ${IREE_ENABLE_THREADING})
+  return()
+endif()
```

Comment:
How is the strings module using threading? Through `iree/base:status`? We should at least add comments to these so we can track it for future cleanup.

In the case of Status, we plan on removing the absl dep and possibly some of the C++ classes, so this check may not be needed once that is resolved.